cheatsheets_draft/GraphQL_Cheat_Sheet.md

```diff
@@ -0,0 +1,257 @@
+
+> This blurb is from nikitastupin. Would be great if we knew the name of this "hint" feature and link to documentation to disable it. I couldn't find anything from a quick Google.
+
+Keep in mind that even if introspection is disabled you can still guess fields by brute forcing them. Furthermore, some GraphQL implementations give you a hint when a field name you provide is similar to an existing field (e.g. you provide `usr` and the response may ask if you meant the valid `user` field instead). You should consider disabling this feature.
+
+***Disable Introspection - Java***
+
+```Java
+GraphQLSchema schema = GraphQLSchema.newSchema()
+    .query(StarWarsSchema.queryType)
+    .fieldVisibility( NoIntrospectionGraphqlFieldVisibility.NO_INTROSPECTION_FIELD_VISIBILITY )
+    .build();
+```
+
+***Disable Introspection & GraphiQL - JavaScript***
+
+```javascript
+app.use('/graphql', graphqlHTTP({
+  schema: MySessionAwareGraphQLSchema,
++ validationRules: [NoIntrospection]
+  graphiql: process.env.NODE_ENV === 'development',
+}));
+```
+
+### IDOR Protection
+
+In general, proper access controls will prevent any IDOR attacks since IDOR merely represents 2 separate authorization issues: [Broken Object Level Authorization](https://github.com/OWASP/API-Security/blob/master/2019/en/src/0xa1-broken-object-level-authorization.md) and [Broken Function Level Authorization](https://github.com/OWASP/API-Security/blob/master/2019/en/src/0xa5-broken-function-level-authorization.md). See the [IDOR prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.html) for a comprehensive rundown of preventing IDOR.
+
+> New section from nikitastupin. Not something I can personally verify with my limited GraphQL experience, so would be great if someone else can check this out.
+
+Sometimes there are `node` or `nodes` or both fields in a query object. They can be used to access objects directly by `ID` which often introduces authorization vulnerabilities. You can check whether your schema has these fields by running this on the command line (assuming that `schema.json` contains your GraphQL schema): `cat schema.json | jq ".data.__schema.types[] | select(.name==\"Query\") | .fields[] | .name" | grep node`. To prevent such vulnerabilities either remove these fields or apply proper authorization checks when objects accessed directly by `ID` as discussed earlier in the "Access Control" section.
+
+## Other Resources
+
+### Tools
+
+- [InQL Scanner](https://github.com/doyensec/inql) - Security scanner for GraphQL. Particularly useful for generating queries and mutations automatically from given schema and them feeding them to scanner.
```

Comment:
InQL isn't really a scanner but rather a tool that helps to explore particular GraphQL API capabilities

Reply:
In the Readme the first title is "InQL Scanner" so I was using that. I think it's more succinct than "Security testing tool" and conveys about the same meaning. I prefer to leave it as is. Any strong arguments/feelings against?

Reply:
Gonna resolve since I haven't heard back. I'm confident in sticking with the wording from the official repo of the tool.